### PR TITLE
refactor Datastore impls so root/basePath creation is user's responsibility

### LINKED ISSUE #21: Datastore impls should not create directories
Their constructors should only check whether the specified directory exists and return a failure if it does not.

See https://github.com/status-im/nim-codex/pull/160#discussion_r933496748.

### PR DESCRIPTION
**Pull request by michaelsbradleyjr**
Closes #21.

### CODE REVIEW

File: datastore/filesystem_datastore.nim [hunk 1]
```diff
@@ -21,18 +21,17 @@ const
 
 proc new*(
   T: type FileSystemDatastore,
-  root = "data"): ?!T =
+  root = getCurrentDir()): ?!T =
```
Comment (michaelsbradleyjr) [inline datastore/filesystem_datastore.nim]: @dryajov `getCurrentDir()` seems like a reasonable default value, but would it be preferable to not have a default value?

File: datastore/sqlite_datastore.nim [hunk 2]
```diff
@@ -172,7 +172,7 @@ proc timestampCol*(
 
 proc new*(
   T: type SQLiteDatastore,
-  basePath = "data",
+  basePath = getCurrentDir(),
```
Comment (michaelsbradleyjr) [inline datastore/sqlite_datastore.nim]: Same question as in my previous comment.

**Comment by emizzle** [comment]
Per meeting: should be a required param to supply a default directory

File: datastore/filesystem_datastore.nim (hunk 1 above)
Comment (michaelsbradleyjr) [reply datastore/filesystem_datastore.nim]: Based on discussion during today's team call, I have changed it so there's no default value for the `root` parameter.

File: datastore/sqlite_datastore.nim (hunk 2 above)
Comment (michaelsbradleyjr) [reply datastore/sqlite_datastore.nim]: Based on discussion during today's team call, I have changed it so there's no default value for the `basePath` parameter.

**Review by dryajov — APPROVED** [review]
LGTM

**Pull request merged**